File: sitoStatic.py
from math import sqrt
from math import ceil

k = 1<<11
isPrime = [True]*k
primes = []
isPrime[0] = False
isPrime[1] = False

for i in range(2, (int)(ceil(sqrt(k)))):
    if(isPrime[i]):
        for j in range(2*i, k, i):
            isPrime[j]=False

for i in range(2, k):
    if(isPrime[i]):
        primes.append(i)

File: SitoStatic.java
import java.util.ArrayList;

class SitoStatic {
    public static void main(String args[]) {
        final int k = 1 << 11;
        Boolean[] isPrime = new Boolean[k];
        ArrayList<Integer> primes = new ArrayList<>();
        isPrime[0] = false;
        isPrime[1] = false;
        for (int i = 2; i < k; i++)
            isPrime[i] = true;
        for (int i = 2; i * i <= k; i++)
            if (isPrime[i] == true)
                for (int j = 2 * i; j < k; j += i)
                    isPrime[j] = false;
        for (int i = 2; i < k; i++)
            if (isPrime[i])
                primes.add(i);
    }
}